pandas/tests/indexing/test_loc.py

```diff
@@ -1563,6 +1563,15 @@ def test_loc_getitem_slice_label_td64obj(self, start, stop, expected_slice):
         expected = ser.iloc[expected_slice]
         tm.assert_series_equal(result, expected)
 
+    def test_loc_getitem_slice_unordered_dt_index(self, frame_or_series):
```

Comment:
can you add the str index example from the OP as well. ping on green.

Reply:
Interestingly enough this does not raise any more. I will open an issue for this. We have to define if ``slice_locs`` should raise too or if we should doo this after calling the function. Is not related to the DateTime fix.

Reply:
#37810